Tests/LibWeb/test-web/Fuzzy.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2025, Jelle Raaijmakers <jelle@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Fuzzy.h"
+
+#include <LibGfx/Bitmap.h>
+
+namespace TestWeb {
+
+// https://web-platform-tests.org/writing-tests/reftests.html#fuzzy-matching
+bool fuzzy_screenshot_match(Gfx::Bitmap const& bitmap_a, Gfx::Bitmap const& bitmap_b, Vector<FuzzyMatch> const& fuzzy_matches)
+{
+    // No fuzzy matching: exact match.
+    auto diff = bitmap_a.diff(bitmap_b);
+    if (fuzzy_matches.is_empty())
+        return diff.identical;
+
+    // Apply each fuzzy match to the diff result.
+    for (auto const& fuzzy_match : fuzzy_matches) {
+        if (fuzzy_match.reference.has_value()) {
+            dbgln("FIXME: Fuzzy match restricted to '{}', but this is not implemented.", fuzzy_match.reference.value());
+            continue;
+        }
+
+        if (fuzzy_match.color_value_error.has_value()) {
+            auto color_error_range = fuzzy_match.color_value_error.value();
+            if (diff.maximum_error < color_error_range.minimum_value || diff.maximum_error > color_error_range.maximum_value)
+                return false;
```

Comment:
It would be nice to output some kind of error at these returns. At the moment it's quite difficult to figure out why a reftest fails.